evalml/pipelines/components/transformers/preprocessing/text_featurizer.py

```diff
@@ -126,16 +130,48 @@ def transform(self, X, y=None):
         if X_nlp_primitives.isnull().any().any():
             X_nlp_primitives.fillna(0, inplace=True)
 
-        X_lsa = self._lsa.transform(X_ww.ww[self._text_columns])
+        X_ww_altered = infer_feature_types(
+            X_ww.ww[self._text_columns].fillna(""),
+            {s: "NaturalLanguage" for s in self._text_columns},
```

Comment:
We use this to keep the natural language type after filling in the nans with empty strings